layers/direct_access.go

```diff
@@ -33,26 +34,27 @@ func (d *DirectAccessLayer) OnRequest(state *httransform.LayerState) error {
 	return nil
 }
 
-func (d *DirectAccessLayer) OnResponse(state *httransform.LayerState, err error) {
+func (d *DirectAccessLayer) OnResponse(ctx *layers.Context, err error) error {
 	if err == errDirectAccess {
```

Comment:
Do you need any assistance here? Please take a look here https://github.com/9seconds/httransform/blob/master/http/execute_test.go and https://github.com/9seconds/httransform/blob/master/http/execute.go it should be straightforward to implement this part.

1. Create a [dialer](https://pkg.go.dev/github.com/9seconds/httransform/v2@v2.0.5/dialers) (this is how we dial to a netloc)
2. Create a [new executor](https://pkg.go.dev/github.com/9seconds/httransform/v2@v2.0.5/executor) (this is how we use dialer to send HTTP/HTTPS requests)
3. If you haven't modified any headers, then `executor(ctx)` would be enough

For direct access, it thins this is going to be enough: https://github.com/9seconds/httransform/blob/master/executor/default_test.go#L90

Reply:
Thank you. Also added `ctx.RequestHeaders.Push()` before and `ctx.ResponseHeaders.Pull()` after the executor to make up for the missing `layerFinishHeaders`